packages/plugin-core/src/test/suite-integ/Telemetry.test.ts

```diff
@@ -0,0 +1,185 @@
+import _ from "lodash";
+import * as vscode from "vscode";
+import { getWS } from "../../workspace";
+import { expect } from "../testUtilsv2";
+import { runLegacyMultiWorkspaceTest, setupBeforeAfter } from "../testUtilsV3";
+import { describe } from "mocha";
+import sinon from "sinon";
+import { SegmentClient } from "@dendronhq/common-server";
+import { setupSegmentClient } from "../../telemetry";
+import { FileTestUtils } from "@dendronhq/common-test-utils";
+import { TestConfigUtils } from "@dendronhq/engine-test-utils";
+import { DisableTelemetryCommand } from "../../commands/DisableTelemetry";
+import { EnableTelemetryCommand } from "../../commands/EnableTelemetry";
+import { DConfig } from "@dendronhq/engine-server";
+
+suite("telemetry", function () {
+  let ctx: vscode.ExtensionContext;
+
+  ctx = setupBeforeAfter(this, {
+    beforeHook: () => {
+      FileTestUtils.mockHome();
+    },
+  });
+
+  describe("configuration", () => {
+    test("enabled by configuration", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        onInit: async ({}) => {
+          const ws = getWS();
+          sinon.stub(vscode.env as any, "isTelemetryEnabled").value(true);
+
+          setupSegmentClient(ws);
+          expect(SegmentClient.instance().hasOptedOut).toBeFalsy();
+
+          sinon.restore();
+          done();
+        },
+      });
+    });
+
+    test("disabled by vscode configuration", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        onInit: async ({}) => {
+          const ws = getWS();
+          sinon.stub(vscode.env as any, "isTelemetryEnabled").value(false);
+
+          setupSegmentClient(ws);
+          expect(SegmentClient.instance().hasOptedOut).toBeTruthy();
+
+          sinon.restore();
+          done();
+        },
+      });
+    });
+
+    test("enabled by vscode configuration, but disabled by workspace configuration", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        preSetupHook: async ({ wsRoot }) => {
+          const config = DConfig.genDefaultConfig();
+          config.noTelemetry = true;
```

Comment:
an easier way to write this is `TestConfigUtils.withConfig` and pass in a callback